lib/rbac/authorizer.rb

```diff
@@ -2,30 +2,26 @@ module Rbac
   class Authorizer
     include Vmdb::Logging
 
-    def self.role_allows?(*args)
-      new.role_allows?(*args)
+    def self.role_allows?(**args)
+      new.role_allows?(**args)
     end
 
-    def role_allows?(options = {})
-      user    = options[:user]
-
-      # 'identifier' comes from the back end (ex: User#role_allows?)
-      # 'feature' comes from the front end (ex: ApplicationHelper#role_allows?)
-      # As this API is a combination of the two, 'feature' is considered
-      # an alias of 'identifier' for legacy purposes.
-      identifier = options[:identifier] || options[:feature]
-
-      # The 'identifiers' option comes from the former User#role_allows_any? API
-      # It should be noted that there may be only ONE caller using 'identifiers'
-      # in Menu::Section, so this option may be changed shortly.
-      identifiers = options[:identifiers]
-
-      any     = options[:any]
-
-      tenant_identifier = MiqProductFeature.current_tenant_identifier(identifier)
-
-      auth = if any.present?
-               user_role_allows_any?(user, :identifiers => (identifiers || [identifier]))
+    # @options option any         - any of the identifiers can match (default: false)
+    # @options option identifier  - comes from the back end (ex: User#role_allows?)
+    # @options option feature     - same as identifier, comes from the front end (ex: ApplicationHelper#role_allows?)
+    # @options option identifiers - same as others
+    #                               This comes from the former User#role_allows_any? API
+    #                               there may be only ONE caller using 'identifiers'
+    #                               in Menu::Section, so this option may be changed shortly.
+    def role_allows?(user:, identifier: nil, feature: nil, identifiers: nil, any: false)
+      identifier ||= feature
+
+      tenant_identifier = MiqProductFeature.current_tenant_identifier(identifier) if identifier.kind_of?(String)
+
+      auth = if user.miq_user_role.nil?
+               false
+             elsif any.present?
+               user_role_allows_any?(user, :identifiers => (identifiers || Array(identifier)))
```

Comment:
Is this related to kwargs?  I think you squashed `any` and identifiers but don't follow this change.

Reply:
```diff
- [identifier]
+ Array(identifier)
```

These are basically the same thing just with some checks incase the user passed `:identifiers => []`, which the original comment hinted we wanted to do.